Tests/kaas/kaas-sonobuoy-go-example-e2e-framework/Dockerfile

```
@@ -1,4 +1,4 @@
-FROM golang:1.17-buster as build
+FROM golang:1.18-buster as build
```

```suggestion
FROM golang:1.23
```

Since this PR is aimed at updating the Golang version, I would suggest upgrading it to the latest Golang version `1.23`.
Additionally, since we are touching this line, the “as build” could be removed as we do not currently use multi-stage builds.

---

Sure, I agree. I just upgraded the Go version to the minimum that worked because I thought there were some restrictions on the versions or something like that since `1.17` is a couple of years old and was used in this Dockerfile only a few months ago.